docs/changes/io.fits/16581.bugfix.rst

```diff
@@ -0,0 +1 @@
+Skip an un-portable hack for Windows if NumPy 2.0 or newer is installed.
```

Comment:
This is in the bugfix section, so as a user I would be very interested in under what circumstances I might encounter this hack. If we cannot figure out a situation where the hack is relevant then the change is not user-facing and should not be in the change log at all.

Reply:
I think the idea is that most readers wouldn't care *but* it's good to have an entry there to signal that we're aware of this change if anyone comes looking for it. Does it make sense ?

Reply:
The entry is so vague that it is impossible for users to know if they should care or not (if they are on Windows).

Reply:
Fair enough. How about: 
> Skip an un-portable hack in for Windows if NumPy 2.0 or newer is installed.
> This fixes a crash that occurred for files oopened via `fits.open(..., mode='update')`

Reply:
Mentioning `fits.open()` and `mode="update"` looks like it might be specific enough, but if the hack can be accessed through the high-level unified I/O interface then that should be mentioned too.

Reply:
I don't know that it can. `fits.open(..., mode="update")` just seemed to be the common denominator for all 4 failing tests we had (see [example logs](https://github.com/astropy/astropy/actions/runs/9536739100/job/26284019167))